BootloaderCorePkg/Include/Library/SmbiosInitLib.h

```diff
@@ -0,0 +1,66 @@
+/** @file
+
+  Copyright (c) 2017 - 2019, Intel Corporation. All rights reserved.<BR>
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php.
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#ifndef __SMBIOS_INIT_LIB_H__
+#define __SMBIOS_INIT_LIB_H__
+
+#include <IndustryStandard/SmBios.h>
+
+
+typedef struct {
+  UINT16    Type;
+  UINT16    Idx;
+  CHAR8     *String;
+} SMBIOS_TYPE_STRINGS;
+
+
+/**
+  Return the pointer to the Smbios table spcified by 'Type'
+
+  @param[in]  Type    Smbios type requested
+
+  @retval             Pointer to the starting of the requested
+                      Smbios type
+
+**/
+VOID *
+FindSmbiosType (
+  IN  UINT8   Type
+  );
+
+/**
+  Append an Smbios Type to an existing set of types
+  Different from AddSmbiosType, which gets called during initial Smbios setup
+
+  @param[in]  TypeData    pointer to the Type Data including strings
+  @param[in]  TypeLength  Size of type data including strings and end terminator (0000)
+
+**/
+VOID
```

Comment:
I think it might be better to return status.
If there is not enough space, it should return error.